misc/python/materialize/output_consistency/expressions/expression_generator.py

```diff
@@ -0,0 +1,149 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+from materialize.output_consistency.data_type.data_provider import DATA_TYPES
+from materialize.output_consistency.data_type.data_type import RawValue
+from materialize.output_consistency.data_type.data_type_group import DataTypeGroup
+from materialize.output_consistency.expressions.expression import Expression
+from materialize.output_consistency.expressions.expression_with_args import (
+    ExpressionWithNArgs,
+)
+from materialize.output_consistency.operations.operation import OperationWithNParams
+from materialize.output_consistency.operations.operation_provider import OPERATION_TYPES
+
+
+class ExpressionGenerator:
+    def generate_expressions(self) -> list[Expression]:
```

Comment:
So you generate all the expressions at once and then possibly pick N of them using `random.choice()` . This may cause an OOM at some point once you add all the stuff that needs testing. You may need to generate just N random expressions rather than generate all and then throw away most of them.